front/components/assistant/conversation/ConversationViewer.tsx

```diff
@@ -357,28 +382,51 @@ export default function ConversationViewer({
           <Spinner variant="color" size="xs" />
         </div>
       )}
-      {messages.map((page) => {
-        return page.messages.map((message) => {
-          return (
-            <MessageItem
-              key={message.sId}
-              conversationId={conversation.sId}
-              hideReactions={hideReactions}
-              isInModal={isInModal}
-              message={message}
-              owner={owner}
-              reactions={reactions}
-              ref={
-                message.sId === prevFirstMessageId
-                  ? prevFirstMessageRef
-                  : undefined
+
+      {groupedMessages.map((group) => {
+        return (
+          // First div is used to apply a background color to the content fragment group.
+          <div
+            className={
+              group[0].type === "content_fragment" ? "bg-structure-50" : ""
```

Comment:
To avoid this check everywhere can we compute a temporary variable something like `isContentFragmentGroup` and based the rendering on this?